### PR TITLE
Major release to the main branch 

### PR DESCRIPTION
**Pull request by t-bello7**
- create and style home and detailed page section
- create the redux store, action, thunks and reducers for the nba data api
- create the search functionality on the home page
- create unit and integration test

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,4 +81,11 @@ Give a ⭐️ if you like this project!
 
 - Microverse.
 - Redux ToolKit Team
--
```
Comment (MahmoudBakr23) [inline README.md:81]: **Required**
- Please give credit for the author of the original design as dictated in the project requirements

File: src/assets/styles/Card.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
+@import "_variables";
+
+.card__container {
```
Comment (MahmoudBakr23) [inline src/assets/styles/Card.scss:3]: **Required**
- Please enhance the design of your app to match the wireframe of the original design. However, the colors are nice.

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @t-bello7 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Great job working with react routes to set up your app navigation
- Well done working with redux actions, reducers, and store along with react hooks to manipulate the state
- Great job working with the API to fetch the data and store it in the store of your app
- Well done designing your app so far
- Great work adding the search feature
- Nice work with jest and test cases
- Descriptive PR
- Correct Linters
- Well done deploying the app and attaching the live version to the README file
- Nice presentation

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (t-bello7) [reply README.md:81]: Thank you @MahmoudBakr23 . Credits were give to the original design in the Resources Used section of the readme

File: src/assets/styles/Card.scss (hunk 2 above)
Comment (t-bello7) [reply src/assets/styles/Card.scss:3]: Thank you @MahmoudBakr23  for the feedback. The design of my applications follows the design guideline of the original design which is the font style( lato nd Gill ) and the blue color.

File: README.md [hunk 3]
```diff
@@ -78,4 +84,10 @@ Give a ⭐️ if you like this project!
 
 - Microverse.
 - Redux ToolKit Team
-- 
+- Nelson Ballhead-App-(Free-PSDs) Design
```
Comment (Ismailco) [inline README.md]: - [ ] Good job adding the credit for the designer as the previous reviewer suggests however Kindly consider adding the correct name of the designer, you can add [Nelson Sakwa](https://www.behance.net/sakwadesignstudio)

File: src/App.jsx [hunk 4]
```diff
@@ -1,11 +1,13 @@
-import './assets/styles/App.scss'
+import { Outlet } from 'react-router-dom'
+import Navbar from './components/Navbar'
 
 const App= () => {
```
Comment (Ismailco) [inline src/App.jsx:4]: - [ ] I don't get any data from your app so I can't really see what the design looks like, and based on your commits after the first review I see that you changed only the README file, so I will suggest modifying the wireframe as the previous reviewer suggest (I don't know if you have issues with the design but logically you didn't change anything from the last time so I will agree with the last reviewer)

Screenshot_20220806_172416

**Review by Ismailco — CHANGES_REQUESTED** [review]
#### Hi @t-bello7, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

giphy


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally


## Suggested changes
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/App.jsx (hunk 4 above)
Comment (t-bello7) [reply src/App.jsx:4]: Hi @Ismailco Thank you for the review. However you can't run the server locally without creating the authentication keys for the NBA API in the environment variable.

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @t-bello7 , 

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

Let me point out: 

- [ ] Project is properly deployed, 
- [ ] The project contains a Home page and a Detail page.
- [ ] The project is a single page application
- [ ] Search bar works as expected 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by t-bello7** [comment]
😄 Thank you @ShoiraTa  @Ismailco @MahmoudBakr23

**Pull request merged**